### PR TITLE
Favorite teams home2.0

### LINKED ISSUE #12: FR11.  Favorite Teams Profile
|FR11.  Favorite Team Profile|
|---|
|**Goal**: The system will allow the user to choose and display their favorite teams.
**Stakeholders:** Users|
| **Description:** A star icon will be displayed next to the team name, the user will be allowed to click on the icon and the desired team will be added to the user's favorites list. |
| **Origin:** During the first meeting.|
| **Version:** 1.0  |  
|**Date:** 02/15/2022  
|  **Priority:** Medium |

### LINKED ISSUE #14: FR9.  A user is able to register with a username and password
FR9.  A user may register a username and password in order to create an account 
|---|
|**Goal**: A user may register a username and password in order to create an account.
**Stakeholders:** Users, Developers|
| **Description:** While not logged in, a user may register for the site with a username and password in order to create an account. |
| **Origin:** Discussed during first meeting.|
| **Version:** 1.0  |
|**Date:** 02/15/2022
|  **Priority:** High |

### LINKED ISSUE #18: FR14.  Favorite Teams Removal
|FR14.  Favorite Teams Removal|
|---|
|**Goal**: The system will allow the user to remove a designated team from their favorites.
**Stakeholders:** Users|
| **Description:** A colored star icon will be displayed next to the team name on the favorites list, the user will be allowed to click on the icon to un-favorite them and the team will be removed from the user's favorites list. |
| **Origin:** During the first meeting.|
| **Version:** 1.0  |  
|**Date:** 02/15/2022  
|  **Priority:** Medium |

### PR DESCRIPTION
**Pull request by dettloffd**
Implemented favoriting and unfavoriting teams on the home page and team page. Favorited teams also now are displayed in the user's profile page.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -44,34 +44,59 @@ app.all('*', (req, res, next) => {
...
 
 /* Home Page  */
-app.get('/', (req, res) => {
-    const getAllTeams = async () => {
+app.get('/', async (req, res) => {
+    const sessionCookie = req.cookies.session || '';
+
+    const response = await axios.get(`${API}/teams.json`); // gets all team name and logo from api
+    const teamData = response.data;
+    const favoriteTeamsArray = [];
```
Comment (muhtacinm) [inline index.js:52]: favoriteTeamsArray doesn't look like it used anymore.

File: index.js [hunk 2]
```diff
@@ -44,34 +44,59 @@ app.all('*', (req, res, next) => {
...
+            });
+
         } catch (error) {
             console.error(error);
         }
-    };
+    }
 
-    getAllTeams();
+    res.render('home', { teamData, favoriteTeamsArray });
```
Comment (muhtacinm) [inline index.js]: favoriteTeamsArray doesn't look like it used anymore.

File: index.js (hunk 1 above)
Comment (muhtacinm) [inline index.js]: favoriteTeamsArray doesn't look like it used anymore.

File: index.js [hunk 3]
```diff
@@ -44,34 +44,59 @@ app.all('*', (req, res, next) => {
...
 });
 
 /* Team Page */
 app.get('/teams/:teamName', async (req, res) => {
     const { teamName } = req.params; // gets teamname
     const sessionCookie = req.cookies.session || '';
     const response = await axios.get(`${API}/${teamName}.json`); // gets team info with players info
+
+    const favoriteTeamsArray = [];
```
Comment (muhtacinm) [inline index.js]: favoriteTeamsArray doesn't look like it used anymore.

File: index.js [hunk 4]
```diff
@@ -83,11 +108,12 @@ app.get('/teams/:teamName', async (req, res) => {
                     }
                 });
             });
+
         } catch (error) {
             console.error(error);
         }
     }
-    res.render('teamPage', { team });
+    res.render('teamPage', { team, favoriteTeamsArray });
```
Comment (muhtacinm) [inline index.js]: favoriteTeamsArray doesn't look like it used anymore.

File: public/css/style.css [hunk 5]
```diff
@@ -1,3 +1,8 @@
+a, a:hover, a:visited, a:active {
```
Comment (muhtacinm) [inline public/css/style.css]: We probably don't want to change all <a> tag attributes. It removes styling for links on the signup page and login page. Probably best to give the <a> tag on the home page an id attribute and change the css for it.

**Review by muhtacinm — APPROVED** [review]
Nice work!

**Review by miqdadhafiz35 — APPROVED** [review]

**Pull request merged**